presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/MapStatistics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata.statistics;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.orc.metadata.statistics.StatisticsHasher.Hashable;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class MapStatistics
+        implements Hashable
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(MapStatistics.class).instanceSize();
+    private final List<MapStatisticsEntry> entries;
+
+    public MapStatistics(List<MapStatisticsEntry> entries)
+    {
+        this.entries = requireNonNull(entries, "entries is null");
+    }
+
+    public List<MapStatisticsEntry> getEntries()
+    {
+        return entries;
+    }
+
+    public long getRetainedSizeInBytes()
+    {
+        long entriesSize = 0;
+        for (MapStatisticsEntry entry : entries) {
+            entriesSize += entry.getRetainedSizeInBytes();
+        }
+        return INSTANCE_SIZE + entriesSize;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MapStatistics that = (MapStatistics) o;
+        return Objects.equals(entries, that.entries);
```

Comment:
Does the two different MapEntry with same contents, but different order different ?

nit: BTW this can be just entries.equals() . entries can never be null, but this way is fine as well.

Reply:
Yes, two MapStatistics with same entries but in different order won't be equal. I called it out in the PR comment asking for opinion.

Reply:
I am fine with the list. This is something internal and developers can manage the complexity, when ready will merge this in.